skorch/utils.py

```diff
@@ -172,6 +175,10 @@ def to_device(X, device):
     if isinstance(X, dict):
         return {key: to_device(val, device) for key, val in X.items()}
 
+    if isinstance(X, Mapping):
+        # dict-like but not a dict
+        return type(X)({key: to_device(val, device) for key, val in X.items()})
```

Comment:
Can this replace the `if isinstance(X, dict)` call above?

Reply:
I removed it.

Reply:
Is this now ready to be merged?